libraries/AP_EFI/AP_EFI_MAV.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_EFI_MAV.h"
+
+#include <GCS_MAVLink/GCS.h>
+
+AP_EFI_MAV *AP_EFI_MAV::singleton;
+
+AP_EFI_MAV::AP_EFI_MAV() {
+    singleton = this;
+}
+
+//Decode MavLink message
+void AP_EFI_MAV::handle_EFI_message(const mavlink_message_t &msg) {
+    mavlink_msg_efi_status_decode(&msg, &efi_data);
```

Comment:
This should populate AP's `EFI_State internal_state` struct. It will then be logged correctly and loaded in to the front end with `copy_to_frontend`.